src/compiler/resolutionCache.ts

```diff
@@ -581,12 +581,13 @@ export function createResolutionCache(resolutionHost: ResolutionCacheHost, rootD
             resolvedModules.push(getResolutionWithResolvedFileName(resolution));
         }
 
-        if (containingSourceFile && resolutionInfo) {
+        // If resolving type reference directive we dont need containingSourceFile to determine if we can use resolutionInfo
+        if (resolutionInfo && (loader === resolveTypeReferenceDirective as unknown || containingSourceFile)) {

```

Comment:
> Will have separate PR that cleans up of these APIs and deprecating the existing APIs

Will this line look less suspicious after an API cleanup?

Reply:
Yeah.. now we need to rely on loader to figure out if we are resolving type ref or modules. The cleanup of API should fix this